openstudiocore/ruby/openstudio/sketchup_plugin/lib/dialogs/SpaceAttributesInterface.rb

```diff
@@ -174,16 +177,27 @@ module OpenStudio
           else
             assigned_const_set = drawing_interface.model_object.setString(3,selected_const_set.to_s)
           end
-
-          # set space's parent zone values
-          # (I could probalby make array of unique values and loop outside of main loop so I don't reset the string over and over for spaces that share zones)
+          
+          # get thermostat object from input name
+          # make thermostat list
+          thermostat_objects = model.getThermostats
+          thermostat_objects.each do |object|
+            if object.name.to_s == thermostat
+              thermostat = object
```

Comment:
@macumber I'll fix that, I also realized I can move this code above the selection loop so it just runs once.